src/_includes/sidebar.html

```diff
@@ -1,94 +1,120 @@
-    <li class="nav-item"><a class="nav-link" href="/testing">Test your app</a></li>
-    <li class="nav-item"><a class="nav-link" href="/testing/debugging">Debug your app</a></li>
-    <li class="nav-item"><a class="nav-link" href="/testing/ui-performance">Performance Profiling</a></li>
-    <li class="nav-item"><a class="nav-link" href="/development/tools/inspector">Inspect your UI</a></li>
-    <li class="nav-item"><a class="nav-link" href="/deployment/android-release">Build and release for Android</a></li>
-    <li class="nav-item"><a class="nav-link" href="/deployment/ios-release">Build and release for iOS</a></li>
-    <li class="nav-item"><a class="nav-link" href="/deployment/fastlane-cd">Continuous deployment with fastlane</a></li>
-    <li class="nav-item"><a class="nav-link" href="/development/tools/upgrading">Upgrade your Flutter install</a></li>
-    <li class="nav-item"><a class="nav-link" href="/development/tools/formatting">Format your source code</a></li>
-  </ul>
+    {%- assign isActive = false -%}
+    {%- if link.urlExactMatch and pageUrl == link.permalink -%}
+      {%- assign isActive = true -%}
+    {%- elsif link.urlExactMatch == nil and pageUrl contains link.permalink -%}
+      {%- assign isActive = true -%}
+    {%- endif -%}
 
-  <li class="nav-item">
-    <a class="nav-link collapsed" data-toggle="collapse" data-target="#detailsMenu" role="button" aria-expanded="false" aria-controls="detailsMenu">More details</a>
-  </li>
+    {% if isActive -%}
+      {% assign class = 'active' -%}
+      {% assign expanded = 'true' -%}
+      {% assign show = 'show' -%}
+    {% else -%}
+      {% assign class = 'collapsed' -%}
+      {% assign expanded = 'false' -%}
+      {% assign show = '' -%}
+    {% endif -%}
+    {% assign id = 'side-bar-' | append: forloop.index -%}
 
-  <ul class="nav flex-column flex-nowrap collapse" id="detailsMenu">
-    <li class="nav-item"><a class="nav-link" href="/resources/faq">FAQ</a></li>
-    <li class="nav-item"><a class="nav-link" href="/resources/technical-overview">Technical overview</a></li>
-    <li class="nav-item"><a class="nav-link" href="https://docs.google.com/presentation/d/1B3p0kP6NV_XMOimRV09Ms75ymIjU5gr6GGIX74Om_DE/edit?usp=sharing">Magic of Flutter slides</a></li>
-    <li class="nav-item"><a class="nav-link" href="https://docs.google.com/presentation/d/1cw7A4HbvM_Abv320rVgPVGiUP2msVs7tfGbkgdrTy0I/edit?usp=sharing">Architecture diagrams</a></li>
-    <li class="nav-item"><a class="nav-link" href="https://www.youtube.com/watch?v=dkyY9WCGMi0">Framework's layered design <i class="fa fa-video-camera" aria-hidden="true"></i></a></li>
-    <li class="nav-item"><a class="nav-link" href="https://www.youtube.com/watch?v=UUfXWzp0-DU">Flutter's rendering pipeline <i class="fa fa-video-camera" aria-hidden="true"></i></a></li>
-  </ul>
+    <li class="nav-item">
+      <a class="nav-link {{class}}" data-toggle="collapse" href="#{{id}}" role="button"
```

Comment:
Instead of `href="#{{id}}"` I would recommend using `data-target="#{{id}}"` so that the `a` tag doesn't render as a link, since it's just an interactive element, but it doesn't take the user anywhere.

Reply:
If you use the `data-target` then when you hover over a top-level entry, the cursor is a `crosshair`. IMHO, that is unintuitive for something that is clickable. That was why I gave the anchor a `href`.

I can drop the `href` attribute, but then we'd need to add a change in cursor. If we keep the `href`, then we'd need to change the color. Which do you prefer? I'd favor the latter. You?

Reply:
@fertolg - merging now so that I can work on iteration 2 of this feature. We can adjust the styles later as needed.